src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java

```diff
@@ -668,6 +655,14 @@ public static void popRegistrationCallback() {
         callbacks.get().pop();
     }
 
+    @DataBoundSetter
+    public synchronized void setApprovedSignatures(String[] signatures) {
```

Comment:
What about classpaths & Co?

Reply:
I think this can be tracked as further improvement separately. Current semantic can be extended in the future to cover this as well. WDYT?